### ISSUE TITLE
__parallelize_with and newsymbol

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#402]
The following Regent snippet:

```
import "regent"

local struct Particle {
  __valid : bool;
}

__demand(__parallel)
task particles_initValidField(r : region(ispace(int1d), Particle))
where writes(r.__valid) do
  __demand(__openmp)
  for e in r do
    e.__valid = false
  end
end

local particles = regentlib.newsymbol()
local p_particles = regentlib.newsymbol()

local DeclSymbols = rquote
  var [particles] = region(ispace(int1d,32), Particle)
  var primColors = ispace(int3d, int3d{2,2,2})
  var [p_particles] = partition(equal, particles, primColors)
end

local InitRegions = rquote
  __parallelize_with p_particles do
    particles_initValidField(particles)
  end
end

local task main()
  [DeclSymbols];
  [InitRegions];
end

regentlib.start(main)
```

fails to compile with:

```
/home/manolis/legion/language/src/regent/symbol_table.t:59: Errors reported during typechecking.
parallelize_with_newsymbol.rg:26: name '$31' is undefined or nil
  __parallelize_with p_particles do
                               ^

stack traceback:
	[C]: in function 'error'
	src/terralib.lua:388: in function 'finishandabortiferrors'
	/home/manolis/legion/language/src/common/report.t:52: in function 'error'
	/home/manolis/legion/language/src/regent/symbol_table.t:59: in function 'lookup'
	/home/manolis/legion/language/src/regent/type_check.t:422: in function 'expr_id'
	/home/manolis/legion/language/src/regent/type_check.t:3613: in function 'fn'
	src/asdl.lua:18: in function 'map'
	/home/manolis/legion/language/src/regent/type_check.t:3611: in function 'fn'
	src/asdl.lua:18: in function 'map'
	/home/manolis/legion/language/src/regent/type_check.t:3162: in function 'block'
	/home/manolis/legion/language/src/regent/type_check.t:3769: in function 'top_task'
	/home/manolis/legion/language/src/regent/type_check.t:3836: in function </home/manolis/legion/language/src/regent/type_check.t:3834>
	/home/manolis/legion/language/src/regent/passes.t:56: in function </home/manolis/legion/language/src/regent/passes.t:48>
	parallelize_with_newsymbol.rg:31: in main chunk
```

**Comment by magnatelee** [comment]
This issue should be fixed by commit c1f194fb.

**Comment by manopapad** [comment]
The issue has been fixed, thanks!
